File: web-fe/src/app/services/answer.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { HttpUtilService } from './http.util.service';
import { Observable } from 'rxjs';
import { environment } from '../environments/environment';
import { AnswerDTO } from '../dtos/answer.dto';

@Injectable({
  providedIn: 'root'
})
export class AnswerService {

  private apiAddAnswers = `${environment.apiBaseUrl}/answers/add`;

  constructor(
    private http: HttpClient,
    private httpUtilService: HttpUtilService
  ) { }

  private getApiConfig() {
    return {
      headers: this.httpUtilService.createHeaders(),
    };
  }

  addAnswers(answers: AnswerDTO[]): Observable<any> {
    return this.http.post(this.apiAddAnswers, answers, this.getApiConfig());
  }
}

File: web-be/src/main/java/com/project/web_be/services/AnswerService.java
package com.project.web_be.services;

import com.project.web_be.dtos.AnswerDTO;
import com.project.web_be.entities.Answer;

import java.util.List;

public interface AnswerService {
    List<Answer> addAnswers(List<AnswerDTO> answerDTOs) throws Exception;
    void deleteAnswer(long id);
}
